ArchiSteamFarm/ArchiWebHandler.cs

```diff
@@ -122,39 +123,63 @@ public sealed class ArchiWebHandler : IDisposable {
 		}
 
 		[ItemCanBeNull]
+		[Obsolete]
 		[PublicAPI]
-		[SuppressMessage("ReSharper", "FunctionComplexityOverflow")]
 		public async Task<HashSet<Steam.Asset>> GetInventory(ulong steamID = 0, uint appID = Steam.Asset.SteamAppID, ulong contextID = Steam.Asset.SteamCommunityContextID, bool? marketable = null, bool? tradable = null, IReadOnlyCollection<uint> wantedRealAppIDs = null, IReadOnlyCollection<uint> unwantedRealAppIDs = null, IReadOnlyCollection<Steam.Asset.EType> wantedTypes = null, IReadOnlyCollection<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity)> wantedSets = null) {
 			if ((appID == 0) || (contextID == 0)) {
 				Bot.ArchiLogger.LogNullError(nameof(appID) + " || " + nameof(contextID));
 
 				return null;
 			}
 
+			try {
+				return await GetInventoryEnumerable(steamID, appID, contextID).Where(
+					item =>
+						(!marketable.HasValue || (item.Marketable == marketable.Value)) &&
+						(!tradable.HasValue || (item.Tradable == tradable.Value)) &&
+						(wantedRealAppIDs?.Contains(item.RealAppID) != false) &&
+						(unwantedRealAppIDs?.Contains(item.RealAppID) != true) &&
+						(wantedTypes?.Contains(item.Type) != false) &&
+						(wantedSets?.Contains((item.RealAppID, item.Type, item.Rarity)) != false)
+				).ToHashSetAsync().ConfigureAwait(false);
+			} catch (IOException) {
+				return null;
+			} catch (Exception e) {
+				Bot.ArchiLogger.LogGenericException(e);
+
+				return null;
+			}
+		}
+
+		[JetBrains.Annotations.NotNull]
+		[PublicAPI]
+		[SuppressMessage("ReSharper", "FunctionComplexityOverflow")]
+		public async IAsyncEnumerable<Steam.Asset> GetInventoryEnumerable(ulong steamID = 0, uint appID = Steam.Asset.SteamAppID, ulong contextID = Steam.Asset.SteamCommunityContextID) {
```

Comment:
I'm wondering whether we should rename the function to something better like `GetInventoryAsync`, it'd be the best if it could just be `GetInventory` but I don't think you can do it due to same declaration as the one before. `GetInventoryAsync` seems good new convention for stuff that generates results in chunks.